### ISSUE TITLE
pipx inject - WARNING: PATH altered to include ./local/pipx/venvs

### ISSUE THREAD

**Issue by diepes** [pypa/pipx#1053]
**Describe the bug**

Getting warning
```
# ansible-lint --version
WARNING: PATH altered to include /root/.local/pipx/venvs/ansible/bin :: This is usually a sign of broken local setup, which can cause unexpected behaviors.
ansible-lint 6.18.0 using ansible-core:2.15.3 ansible-compat:4.1.9 ruamel-yaml:0.17.32 ruamel-yaml-clib:0.2.7
```

**How to reproduce**

Debian container
```
RUN  pipx install --include-deps ansible \
  && pipx inject ansible --include-apps ansible-lint
```

**Expected behavior**

No warning.

The path mentioned in the warning is strange "/root/.local/pipx/venvs/ansible/bin".
I assume this is because i "pipx inject" the ansible-lint in to the ansible venv.

my PATH does include "/root/.local/bin",  but not the path in the warning.

**Comment by uranusjr** [comment]
This warning is from `ansible-lint` so you’ll need to talk to them. I’m not sure why they think this is a sign of broken setup.

**Comment by diepes** [comment]
@uranusjr to be sure I understand you.

when you say "pipx indeed adds this to PATH" do you mean

a: "/root/.local/bin"
or
b: "/root/.local/pipx/venvs/ansible/bin"

**Comment by chriscroome** [comment]
The [linked fix](https://github.com/ansible/ansible-lint/pull/3720) has been released with [ansible-lint 16.19.0](https://github.com/ansible/ansible-lint/releases/tag/v6.19.0) and I have this version installed using `pipx` however I still have the warning:

```bash
ansible-lint --version
WARNING: PATH altered to include /home/chris/.local/pipx/venvs/ansible/bin :: This is usually a sign of broken local setup, which can cause unexpected behaviors.
ansible-lint 6.19.0 using ansible-core:2.15.3 ansible-compat:4.1.10 ruamel-yaml:0.17.32 ruamel-yaml-clib:0.2.7
```

**Comment by uranusjr** [comment]
pipx (on Linux) symlinks the executable from `/root/.local/pipx/venvs/ansible/bin/ansible-link` into `/root/.local/bin`. The latter is obviously always in PATH (for the executable to be found), but since pipx installs ansible-link in a virtual environment (specifically, in `/root/.local/pipx/venvs/ansible`), adding the virtual environment’s `bin` to PATH when running ansible-lint is allowed per PEP 405. The problem remainds at why exactly ansible-lint decides this somehow indicates a broken setup.

**Comment by uranusjr** [comment]
Closing since there’s nothing pipx can do at the moment, from our perspective the setup does not seem to be out of ordinary.

**Comment by chrysle** [comment]
> The [linked fix](https://github.com/ansible/ansible-lint/pull/3720) has been released with [ansible-lint 16.19.0](https://github.com/ansible/ansible-lint/releases/tag/v6.19.0) and I have this version installed using `pipx` however I still have the warning:

Sorry, I was fixing the bug in the wrong place :-(

Will open a new pull request shortly.
